File: Design-Patterns/Builder-Design-Pattern/ts/js/DentistBuilder/DentistBuilder.ts
import { Dentist } from './Dentist';

/*
 * File Created: Saturday, 24th August 2019 12:58:56 pm
 * Author: yousef
 * -----
 * Copyright © 2019
 */

export interface DentistBuilder {
	buildPersonalInformation(): void;
	buildEducation(): void;
	buildAddress(): void;
	buildSpecializationArea(): void;
	getDentist(): Dentist;
}

File: Design-Patterns/Builder-Design-Pattern/java/src/builderdesignpattern/DentistBuilder.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package builderdesignpattern;

import java.util.ArrayList;

import dentist.Address;
import dentist.Education;
import dentist.PersonalInformation;

/**
 *
 * @author yousef
 */
public interface DentistBuilder {
    
    public void buildPersonalInformation();
    public void buildEducation();
    public void buildAddress();
    public void buildSpecializationArea();
    public Dentist getDentist(); 

}
